src/d3a/models/strategy/area_agents/one_sided_alternative_pricing_agent.py

```diff
@@ -66,8 +68,17 @@ def _buy_energy_alternative_pricing_schemes(self, area):
                                "An Error occurred while buying an offer")
 
     def event_tick(self, *, area):
-        if area.current_tick_in_slot - self.MIN_SLOT_AGE >= 0 and \
+        if area.current_tick_in_slot >= self.MIN_SLOT_AGE and \
                 ConstSettings.IAASettings.AlternativePricing.PRICING_SCHEME != 0:
             self._buy_energy_alternative_pricing_schemes(area)
 
-        super().event_tick(area=area)
+    def event_market_cycle(self):
+        if ConstSettings.IAASettings.AlternativePricing.PRICING_SCHEME != 0:
+            energy_per_slot = int(sys.maxsize)
+            energy_rate = self.owner.config.market_maker_rate[self.lower_market.time_slot_str]
```

Comment:
In case of no-scheme, the rate should be zero. So, shouldn't this be based on pricing scheme rather than equal to MMR?

Reply:
This code is referring to the trade from the IAA to the house loads. Therefore, even in the no-scheme, the rate should not be zero for this, it should be the market maker rate, since the IAA does not sell energy for free to the house devices.

Reply:
Then, I guess we should simply remove the no scheme option?

Reply:
Not really, because the no-scheme option refers to the trade from the device to the IAA, it denotes that the energy surplus of the house is given away for free to the TSO.